src/_pytest/main.py

```diff
@@ -469,7 +470,7 @@ def _perform_collect(self, args, genitems):
             return items
 
     def collect(self):
-        for parts in self._initialparts:
+        for parts in self._initialparts:  # noqa: F402
```

Comment:
I do not really like this (it is needed because of the imported path being shadowed).
Suggestions?

Reply:
for initialpart in self._initialparts?